### DISCUSSION TITLE
Returned datetime is not accurate

### DISCUSSION THREAD

**Discussion by shambakey1** [GeoscienceAustralia/gnssanalysis#14] [Q&A] [answered]
Hi

I'm trying to convert the datetime parameter from the multi index column, but the conversion to "year-month-day hour-min-sec" is not accurate with me. I'm not sure what I'm doing wrong. For example, for the file "https://cddis.nasa.gov/archive/gnss/products/ionex/2022/001/igsg0010.22i.Z", the epoch is for date 2022-01-01. I read the file using the following syntax:
```
import gnssanalysis as gns
import datetime

d=gns.gn_io.ionex.read_ionex(<file path>)
datetime.datetime.fromtimestamp(d[0].index.values[0][0])
```
The output of the previous command is 
```
datetime.datetime(1992, 1, 1, 14, 0)
```
Which is NOT 2022. I tried to change the command to
```
datetime.datetime.utcfromtimestamp(d[0].index.values[0][0])
```
But still the same result.
I wonder what I'm doing wrong.

Regards

**Comment by seballgeyer** [comment, marked as answer]
Hi

You will need to use the following function 
```python
gns.gn_datetime.j2000_to_pydatetime(d[0].index[0][0])
```

this will return 
```
datetime.datetime(2022, 1, 1, 0, 0)
```

inside `gnssanalysis` datetime are stored as number of seconds since J2000 (01-01-2000T12:00:00) whereas
the datetime library has its 0 being `01-01-1970T00-00-00`, the difference of origin time explain the difference in results. 

Best regards

Sébastien

**Reply by shambakey1** [reply]
Thank you very much. This solved it
